sign/rpmgensig.c

```diff
@@ -307,8 +337,8 @@ static int runGPG(sigTarget sigt, const char *sigfile)
     Fclose(fnamedPipe);
     fnamedPipe = NULL;
 
-    (void) waitpid(pid, &status, 0);
-    pid = 0;
+    (void) waitpid(gpgPid, &status, 0);
```

Comment:
Hmm, what if we get a SIGCHLD in the meanwhile? That'd set gpgPid to 0, which would turn the above into waitpid(0, ...) which doesn't seem right.